### ISSUE TITLE
Verb phrase finder doesn't look for prepositions or direct objects

### ISSUE THREAD

**Issue by ahalterman** [openeventdata/UniversalPetrarch#26]
In looking for parts of the verb phrase, UniversalPetrarch only looks for negations, compounds, and adverbial clause modifiers. It doesn't look for prepositions ("fired AT targets") or direct objects ("chanted SLOGANS). ([code](https://github.com/openeventdata/UniversalPetrarch/blame/2c655c29260062a66c8901c4bfb246f56b28ffa7/UniversalPetrarch/PETRgraph.py#L752)) Prepositions and direct objects are major parts of verb patterns (indeed, they're specially marked in the dictionaries to make matching possible). This lack of code rules out using any verb dictionary entries that include any direct objects or prepositions and would explain why the Petr2 dictionaries were working so badly.

**Comment by JingL1014** [comment]
In current coder, the way to handle prepositional phrase is a little bit different from PETR2 implementation. As mentioned in #27, in step 1, the action in the "fired AT targets" is extracted as "fired at", and prepositions are handled inside verb phrase [(code)](https://github.com/openeventdata/UniversalPetrarch/blob/d32042225c75299ce73375fbc8f0f425a6619e2e/UniversalPetrarch/PETRgraph.py#L1277)

**Comment by ahalterman** [comment]
Looking at the code, that seems to handle the comparison between the extracted verb phrase and the *dictionaries*. What I'm concerned about is the ability of UniversalPetrarch to extract preps and dobjs in the verb phrase from the text to begin with. If it can't extract them correctly, then it won't have any of them to compare against the dictionary in that code chunk. Am I missing something?

**Comment by ahalterman** [comment]
Just checking in on this again, especially since the preposition issue came up in Arabic. Is the code indeed extracting prepositional phrases coming off the verb? And am I right in thinking that that code concerns pattern matching, not phrase extraction?

**Comment by JingL1014** [comment]
Yes. Now the coder concerns the prepositions coming off the verb to do the pattern matching. It matches the entire verb phrases (the verb and a sequence of other chunks such as a noun phrase or a prepositional phrase that follow the verb)

**Comment by ahalterman** [comment]
Great! Can you link to the part of the code that implements it, just so I can make sure I understand where it's doing it?

**Comment by JingL1014** [comment]
The code can be found in function match_lower() starting [from line 1660 to line 1700](https://github.com/openeventdata/UniversalPetrarch/blob/f2c57d322ff31b184bbf291460e56d49630507d7/UniversalPetrarch/PETRgraph.py#L1660-L1700)
